docs: add newbie guideline and template (#47834)
freeCodeCamp/freeCodeCamp: docs/how-to-open-a-pull-request.md

A pull request (PR) enables you to send changes from your fork on GitHub to freeCodeCamp.org's main repository. Once you are done making changes to the code, you can follow these guidelines to open a PR.

We expect our contributors to be aware of the process specific to this project. Following the guidelines religiously earns you the respect of fellow maintainers and saves everyone time.

Some examples of this are:

> 1. Do not edit off of GitHub–while you can, it's not a good idea.
> 2. Make sure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously.
> 3. Use the correct way to link PRs in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.
> 4. Use mentions in comments or review requests to a minimum or not. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.

Reviewer: ```suggestion
1. Do not edit files directly through GitHub – while you can, it's not a good idea.
2. Make sure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously.
3. Use the correct way to link issues in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.
4. Use mentions in comments or review requests to a minimum or not. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.
```

A pull request (PR) enables you to send changes from your fork on GitHub to freeCodeCamp.org's main repository. Once you are done making changes to the code, you can follow these guidelines to open a PR.

We expect our contributors to be aware of the process specific to this project. Following the guidelines religiously earns you the respect of fellow maintainers and saves everyone time.

Some examples of this are:

1. Do not edit files directly through GitHub – while you can, it's not a good idea.
2. Make sure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously.
3. Use the correct way to link issues in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.

> 4. Use mentions in comments or review requests to a minimum or not. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.

Reviewer 2: ```suggestion
4. Keep mentions and review requests to a minimum. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.
```

Minor suggestion. That first sentence confused me a little - keep it to a minimum or not?

> 4. Use mentions in comments or review requests to a minimum or not. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.

Reviewer 2: ```suggestion
4. Keep mentions and review requests to a minimum. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.
```

Minor suggestion. That first sentence confused me a little - keep it to a minimum or not?

5. Do not work directly off your `main` branch - create a new branch for the changes you are working on.

> [!NOTE]

> > Your PR should be targeting changes to the curriculum in English only. Read [this guide](index.md#translations) instead for contributing to translations.

Author: IMHO, it's pretty apparent already, isn't it? On the other hand, I made the change because we saw people changing files that they should not be. 

I have never seen a PR with a description that was not in English. Even on the off-chance, it did, we could easily handle that as maintainers.

> > Your PR should be targeting changes to the curriculum in English only. Read [this guide](index.md#translations) instead for contributing to translations.

Reviewer 3: ```suggestion
> Your PR should be targeting changes to the English curriculum only. Read [this guide](index.md#translations) instead for contributing to translations.
```

Maybe? in case that "in english" is still about the language of the PR?

1. Once the edits have been committed, you will be prompted to create a pull request on your fork's GitHub Page.

   <details>

>    <summary>See a screenshot</summary>

Author: ```suggestion
   <summary>See screenshot</summary>
```

   ![Image - Compare & pull request prompt on GitHub](https://contribute.freecodecamp.org/images/github/compare-pull-request-prompt.png)

   </details>

2. By default, all pull requests should be against the freeCodeCamp main repo, `main` branch.

   Make sure that your Base Fork is set to freeCodeCamp/freeCodeCamp when raising a Pull Request.

   <details>

>    <summary>See a screenshot</summary>

Author: ```suggestion
   <summary>See screenshot</summary>
```
